docs/dev/designs/RestServer.md

```diff
@@ -0,0 +1,176 @@
+# REST server
+
+Siodb must provide an integrated REST server that can work in parallel with the SQL front-end.
+The server will be a distinct process called `siodb_rest_server`.
+There will be new parameters to indicate the ports numbers on which the process will
+listen:
+
+- `rest_server.ipv4_http_port`. Default port `50080`. `0` means do not listen.
+- `rest_server.ipv4_https_port`. Default port `50443`. `0` means do not listen.
+- `rest_server.ipv6_http_port`. Default port `50080`. `0` means do not listen.
+- `rest_server.ipv6_https_port`. Default port `50443`. `0` means do not listen.
+
+The process will convert the REST queries into the IOMgr engine protobuf
+message (From REST to SQL).
+
+It would only work per table and allow users to push/retrieve data through REST.
+
+## Third-party libraries
+
+Siodb will use Oat++ (https://github.com/oatpp/oatpp) for the implementation of the REST server
+in the `siodb_rest_server` process.
+
+Also, we would leverage the JSON API from Protobuf as much as possible for conversion between
+message to JSON. E.g. when sending the IOMgr response to the client.
+
+## Authorization
```

Comment:
must be Authentication
Authentication is process of establishing identity - that's what we achieve by providing token
Authorization is different thing - it is decision making, whether certain user is allowed to perform certain action. Authorization can only happen when user identity is already established. Authorization is performed by database using access permissions to database objects (we do not support that yet).